development/image-builder/main.go

```diff
@@ -221,13 +229,109 @@ func runBuildJob(o options, vs Variants) error {
 			errs = append(errs, fmt.Errorf("job %s ended with error: %w", variant, err))
 			fmt.Printf("Job '%s' ended with error: %s.\n", variant, err)
 		} else {
+			err := signImages(&o, destinations)
+			if err != nil {
+				return fmt.Errorf("sign encountered error: %w", err)
+			}
 			fmt.Println("Successfully built image:", strings.Join(destinations, ", "))
 			fmt.Printf("Job '%s' finished successfully.\n", variant)
 		}
 	}
 	return errutil.NewAggregate(errs)
 }
 
+func signImages(o *options, images []string) error {
+	// use o.orgRepo as default value since someone might have loaded is as a flag
+	orgRepo := o.orgRepo
+	if o.isCI {
+		// try to extract orgRepo from Prow-based env variables
+		org := os.Getenv("REPO_OWNER")
+		repo := os.Getenv("REPO_NAME")
+		if len(org) > 0 && len(repo) > 0 {
+			// assume this is our variable since both variables are present
+			orgRepo = org + "/" + repo
+		}
+	}
+	if len(orgRepo) == 0 {
+		return fmt.Errorf("'orgRepo' cannot be empty")
+	}
+	sig, err := getSignersForOrgRepo(o, orgRepo)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Start signing images", strings.Join(images, ","))
+	var errs []error
+	for _, s := range sig {
+		err := s.Sign(images)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("sign error: %w", err))
+		}
+	}
+	return errutil.NewAggregate(errs)
+}
+
+// getSignersForOrgRepo fetches all signers for a repository
+// It fetches all signers from '*' and specific org/repo combo.
+func getSignersForOrgRepo(o *options, orgRepo string) ([]sign.Signer, error) {
+	c := o.SignConfig
+	if len(c.EnabledSigners) == 0 {
+		// no signers enabled. no need to gather signers
+		return nil, nil
+	}
+	var enabled StrList
+
+	defaultSigners := c.EnabledSigners["*"]
+	orgRepoSigners := c.EnabledSigners[orgRepo]
+	for _, s := range append(defaultSigners, orgRepoSigners...) {
+		enabled.Add(s)
+	}
+	fmt.Println("sign images using services", strings.Join(enabled.List(), ","))
+	var signers []sign.Signer
+	for _, sc := range c.Signers {
+		if enabled.Has(sc.Name) {
+			s, err := sc.Config.NewSigner()
+			if err != nil {
+				return nil, fmt.Errorf("signer init: %w", err)
+			}
+			signers = append(signers, s)
+		}
+	}
+	return signers, nil
+}
+
+// StrList implements list of strings as a map
+// This implementation allows getting unique values when merging multiple maps into one
+// (@Ressetkk): We should find better place to move that code
```

Comment:
will you keep this internal comment?

Reply:
Yes. I've left every comment in the code intentionally, so when time comes we could refactor that to separate package. I just didn't have idea where to put it.